packages/api/app/Controllers/Http/UserController.js

```diff
@@ -82,15 +81,15 @@ class UserController {
 		if (role) {
 			const newUserRole = await Role.getRole(role);
 			if (upUser.role_id !== newUserRole.id) {
-				upUser.role().dissociate();
+				await upUser.role().dissociate();
 				await upUser.role().associate(newUserRole);
 			}
 		}
 
 		if (permissions) {
 			await upUser.permissions().detach();
 			const permissionCollection = await Permission.query()
-				.whereIn('permission', permissions)
+				.whereIn({ permissions })
```

Comment:
Acho que você precisa renomear a variável para `permission` ao inves de `permissions` veja que `.whereIn('permission', permissions)` é diferente de `.whereIn({ permissions }) => .whereIn({ permissions: permissions })`

Reply:
Exato, isso aqui também indica que os testes não estão cobrindo tudo! pois os testes ainda estão passando!

Reply:
Eu não sei se entendi direito @marcusmota , mas nesse caso o whereIn é como se fosse assim:
`select * from sabia.permissions where permission in ([permissions]);`
por isso `.wherein('permission',permissions)`

Reply:
@nicholasio com relação aos testes, mesmo que eu não faça eu testo manualmente as rotas no ignominia para ver o comportamento delas, nesse caso eu mando um array de permissions e associo aos usuários diretamente.

Reply:
Meu ponto aqui lelek foi com relação a isso https://alligator.io/js/object-property-shorthand-es6/

ao passar `{ permissions }` como parametro do `.whereIn` você mudou a lógica do seu código, pois antes você tinha `.whereIn('permission', permissions)` veja que o segundo parametro é no plural ao inves do singular, como você mudou para a sintaxe nova, o código ficou `.whereIn({ permissions })` o que não irá funcionar pois `permissions` é diferente de `permission` (que é o campo que existe na base de dados e não permissions), minha sugestão foi mudar o nome da variável para o singular e.g permission, assim você consegue passar como parametro e sua função continua funcionando (pois com a mudança que você fez, ela quebrou), ficando assim `.whereIn({ permission })`

Reply:
outra opção é fazer assim: `.whereIn({ permissions: permission })`

Reply:
Algumas considerações acerca desses pontos:
1. @marcusmota o meu código original não foi mudado, essa notação .whereIn('permission', permissions) está correta, foi testada, o ponto aqui é se devo deixar assim ou usar a notação de objetos que @nicholasio sugeriu `.whereIn({ permission: permissions })`. Aproveito para mencionar também que essa notação nem sempre funciona, por exemplo:
```javascript
getOwner() {
		return this.users()
			.wherePivot('role', 'OWNER')
			.first();
	}
```
Se eu mudar para:
getOwner() {
		return this.users()
			.wherePivot({role:'OWNER'})
			.first();
	}
Não funciona. Acredito que essa notação só funcione no .where()

Reply:
Acho que na prática dificilmente iriamos filtrar por mais de um campo no `wherePivot` então podemos usar a notação normal.

Já para o `where` principalmente, acho que seria interessante usar a notação de objetos pra facilitar adicionais mais filtros no futuro caso seja necessário.

Reply:
Só para complementar aquestão do erro aqui no `whereIn({permissions})` estou testando essa branch (e fazendo algumas pequenas modificações e essa parte aqui de fato está falhando (escrevi um novo teste)

![image](https://user-images.githubusercontent.com/6104632/84451551-c1022000-ac29-11ea-864d-b2e5b6fd79a0.png)

Estou consertando isso aqui mas só pra esclecer o ponto aqui :)

Reply:
E só pra confirmar que a notação de objeto só funciona com `.where` mesmo